// shiki/dist/bundle-web.mjs
import { bundledThemes } from './themes.mjs';
export { bundledThemesInfo } from './themes.mjs';
export { g as getWasmInlined } from './wasm-dynamic-K7LwWlz7.js';
import { createdBundledHighlighter, createSingletonShorthands, warnDeprecated } from '@shikijs/core';
export * from '@shikijs/core';
import { createOnigurumaEngine } from '@shikijs/engine-oniguruma';

const bundledLanguagesInfo = [
  {
    "id": "angular-html",
    "name": "Angular HTML",
    "import": () => import('./angular-html-CZb8z5sS.js').then(function (n) { return n.f; })
  },
  {
    "id": "angular-ts",
    "name": "Angular TypeScript",
    "import": () => import('./angular-ts-DNsftJy6.js')
  },
  {
    "id": "astro",
    "name": "Astro",
    "import": () => import('./astro-0OknWUVH.js')
  },
  {
    "id": "blade",
    "name": "Blade",
    "import": () => import('./blade-MNtB4jtG.js')
  },
  {
    "id": "c",
    "name": "C",
    "import": () => import('./c-DuaBppJD.js')
  },
  {
    "id": "coffee",
    "name": "CoffeeScript",
    "aliases": [
      "coffeescript"
    ],
    "import": () => import('./coffee-CTnX49HY.js')
  },
  {
    "id": "cpp",
    "name": "C++",
    "aliases": [
      "c++"
    ],
    "import": () => import('./cpp-BQUibumd.js')
  },
  {
    "id": "css",
    "name": "CSS",
    "import": () => import('./css-DSIbPk70.js')
  },
  {
    "id": "glsl",
    "name": "GLSL",
    "import": () => import('./glsl-DtKEILuY.js')
  },
  {
    "id": "graphql",
    "name": "GraphQL",
    "aliases": [
      "gql"
    ],
    "import": () => import('./graphql-B9eBS01g.js')
  },
  {
    "id": "haml",
    "name": "Ruby Haml",
    "import": () => import('./haml-H4-fCAxd.js')
  },
  {
    "id": "handlebars",
    "name": "Handlebars",
    "aliases": [
      "hbs"
    ],
    "import": () => import('./handlebars-DA5vNYKM.js')
  },
  {
    "id": "html",
    "name": "HTML",
    "import": () => import('./html-BzzzYAaY.js')
  },
  {
    "id": "html-derivative",
    "name": "HTML (Derivative)",
    "import": () => import('./html-derivative-tYqIQUr1.js')
  },
  {
    "id": "http",
    "name": "HTTP",
    "import": () => import('./http-DHuNab_l.js')
  },
  {
    "id": "imba",
    "name": "Imba",
    "import": () => import('./imba-DBnhIRK-.js')
  },
  {
    "id": "java",
    "name": "Java",
    "import": () => import('./java-IrbiBQof.js')
  },
  {
    "id": "javascript",
    "name": "JavaScript",
    "aliases": [
      "js"
    ],
    "import": () => import('./javascript-CqcyNC-b.js')
  },
  {
    "id": "jinja",
    "name": "Jinja",
    "import": () => import('./jinja-BZ1B-IEV.js')
  },
  {
    "id": "jison",
    "name": "Jison",
    "import": () => import('./jison-DimN_uwt.js')
  },
  {
    "id": "json",
    "name": "JSON",
    "import": () => import('./json-DJpGb4aK.js')
  },
  {
    "id": "json5",
    "name": "JSON5",
    "import": () => import('./json5-xiH8NrHm.js')
  },
  {
    "id": "jsonc",
    "name": "JSON with Comments",
    "import": () => import('./jsonc-BcVAFjjj.js')
  },
  {
    "id": "jsonl",
    "name": "JSON Lines",
    "import": () => import('./jsonl-B88-IDXI.js')
  },
  {
    "id": "jsx",
    "name": "JSX",
    "import": () => import('./jsx-iOOXFjcb.js')
  },
  {
    "id": "julia",
    "name": "Julia",
    "aliases": [
      "jl"
    ],
    "import": () => import('./julia-nmggZTTX.js')
  },
  {
    "id": "less",
    "name": "Less",
    "import": () => import('./less-_BDsxzii.js')
  },
  {
    "id": "markdown",
    "name": "Markdown",
    "aliases": [
      "md"
    ],
    "import": () => import('./markdown-UKwuWRsH.js')
  },
  {
    "id": "marko",
    "name": "Marko",
    "import": () => import('./marko-BNq1UnDA.js')
  },
  {
    "id": "mdc",
    "name": "MDC",
    "import": () => import('./mdc-BSDAIlkO.js')
  },
  {
    "id": "mdx",
    "name": "MDX",
    "import": () => import('./mdx-BeSu8qL9.js')
  },
  {
    "id": "php",
    "name": "PHP",
    "import": () => import('./php-DNfc_tIv.js')
  },
  {
    "id": "postcss",
    "name": "PostCSS",
    "import": () => import('./postcss-Cnds3vBF.js')
  },
  {
    "id": "pug",
    "name": "Pug",
    "aliases": [
      "jade"
    ],
    "import": () => import('./pug-ZlA_lcIS.js')
  },
  {
    "id": "python",
    "name": "Python",
    "aliases": [
      "py"
    ],
    "import": () => import('./python-CzA5y1OT.js')
  },
  {
    "id": "r",
    "name": "R",
    "import": () => import('./r-DvMntmLc.js')
  },
  {
    "id": "regexp",
    "name": "RegExp",
    "aliases": [
      "regex"
    ],
    "import": () => import('./regexp-BjgEfPDk.js')
  },
  {
    "id": "sass",
    "name": "Sass",
    "import": () => import('./sass-BkPx2CHr.js')
  },
  {
    "id": "scss",
    "name": "SCSS",
    "import": () => import('./scss-BNstWnNl.js')
  },
  {
    "id": "shellscript",
    "name": "Shell",
    "aliases": [
      "bash",
      "sh",
      "shell",
      "zsh"
    ],
    "import": () => import('./shellscript-DKyerd8X.js')
  },
  {
    "id": "sql",
    "name": "SQL",
    "import": () => import('./sql-CEDmkhXj.js')
  },
  {
    "id": "stylus",
    "name": "Stylus",
    "aliases": [
      "styl"
    ],
    "import": () => import('./stylus-CqSgIeWj.js')
  },
  {
    "id": "svelte",
    "name": "Svelte",
    "import": () => import('./svelte-BiBgdEeW.js')
  },
  {
    "id": "ts-tags",
    "name": "TypeScript with Tags",
    "aliases": [
      "lit"
    ],
    "import": () => import('./ts-tags-CgPgz6sC.js')
  },
  {
    "id": "tsx",
    "name": "TSX",
    "import": () => import('./tsx-DIYQIvC7.js')
  },
  {
    "id": "typescript",
    "name": "TypeScript",
    "aliases": [
      "ts"
    ],
    "import": () => import('./typescript-B0pP7Feb.js')
  },
  {
    "id": "vue",
    "name": "Vue",
    "import": () => import('./vue-CvLK70zP.js')
  },
  {
    "id": "vue-html",
    "name": "Vue HTML",
    "import": () => import('./vue-html-pnfxujsS.js')
  },
  {
    "id": "wasm",
    "name": "WebAssembly",
    "import": () => import('./wasm-DXrXlAKv.js')
  },
  {
    "id": "wgsl",
    "name": "WGSL",
    "import": () => import('./wgsl-CwM4Py0K.js')
  },
  {
    "id": "xml",
    "name": "XML",
    "import": () => import('./xml-DJOQn5Ly.js')
  },
  {
    "id": "yaml",
    "name": "YAML",
    "aliases": [
      "yml"
    ],
    "import": () => import('./yaml-CirVcHoZ.js')
  }
];
const bundledLanguagesBase = Object.fromEntries(bundledLanguagesInfo.map((i) => [i.id, i.import]));
const bundledLanguagesAlias = Object.fromEntries(bundledLanguagesInfo.flatMap((i) => i.aliases?.map((a) => [a, i.import]) || []));
const bundledLanguages = {
  ...bundledLanguagesBase,
  ...bundledLanguagesAlias
};

const createHighlighter = /* @__PURE__ */ createdBundledHighlighter({
  langs: bundledLanguages,
  themes: bundledThemes,
  engine: () => createOnigurumaEngine(import('shiki/wasm'))
});
const {
  codeToHtml,
  codeToHast,
  codeToTokensBase,
  codeToTokens,
  codeToTokensWithThemes,
  getSingletonHighlighter,
  getLastGrammarState
} = /* @__PURE__ */ createSingletonShorthands(
  createHighlighter
);
const getHighlighter = (options) => {
  warnDeprecated("`getHighlighter` is deprecated. Use `createHighlighter` or `getSingletonHighlighter` instead.");
  return createHighlighter(options);
};

export { bundledLanguages, bundledLanguagesAlias, bundledLanguagesBase, bundledLanguagesInfo, bundledThemes, codeToHast, codeToHtml, codeToTokens, codeToTokensBase, codeToTokensWithThemes, createHighlighter, getHighlighter, getLastGrammarState, getSingletonHighlighter };
